src/LanguageServer/Impl/Implementation/Server.cs

```diff
@@ -625,6 +625,7 @@ private sealed class ReloadModulesQueueItem : IAnalyzable {
                         LogMessage(MessageType.Error, t.Exception.Message);
                         return;
                     }
+                    _shutdownCts.Token.ThrowIfCancellationRequested();
```

Comment:
It's enough to call it either here or in `OnDocumentChangeProcessingCompleteAsync`.